### PR TITLE
Refactor argument parsing in TensileCreateLibrary

### PR DESCRIPTION
**Pull request by bstefanuk**
Previously, argument parsing consumed 117 LOC in *TensileCreateLibrary* (TCL), this PR moves the argument parsing logic into a module, updates help strings, default behaviour, and adds other utilities for displaying deprecation warning.

**Outcomes**
- New action called `DeprecatedOption` throws a warning if a deprecated option is used.

**Notable changes**
- Parsing was moved into a module under *TensileCreateLib/*, this is intended as a temporary name for the module until the *TensileCreateLibrary* main refactor is complete, at which point the program entrypoint will be contained within the module, and it can be renamed to *TensileCreateLibrary/*. Current usage of imports throughout the project obfuscates the use of TCL as a file vs. directory module, and thereby we can't use both a file and folder that have the same module name without larger import refactors that are beyond the scope of this ticket.

### CODE REVIEW

File: Tensile/TensileCreateLibrary.py [hunk 1]
```diff
@@ -1154,240 +1544,193 @@ def writeMasterFile(libraryPath: Path, format: str, naming: dict, name: str, lib
...
-
-  arguments["CpuThreads"] = args.CpuThreads
-  arguments["PrintLevel"] = args.PrintLevel
-  arguments["IgnoreAsmCapCache"] = args.IgnoreAsmCapCache
-  arguments["WriteMasterSolutionIndex"] = args.WriteMasterSolutionIndex
-
-  for key, value in args.global_parameters:
-    arguments[key] = value
-  
-  globalParameters["PrintLevel"] = arguments["PrintLevel"]
```
Comment (ellosel) [inline Tensile/TensileCreateLibrary.py:1275]: What happened to this?

File: .jenkins/common.groovy [hunk 2]
```diff
@@ -55,6 +55,10 @@ def runCompileCommand(platform, project, jobName, boolean debug=false)
             export TENSILE_COMPILER=${compiler}
             export HIPCC_COMPILE_FLAGS_APPEND='-O3 -Wno-format-nonliteral -parallel-jobs=4'
 
+            pwd
+            cat requirements.txt
+            pip3 install -r `pwd`/requirements.txt
+
```
Comment (bstefanuk) [inline .jenkins/common.groovy:61]: Delete

File: Tensile/TensileCreateLib/ParseArguments.py [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+        help="Disables short files names.",
+    )
+    parser.add_argument(
+        "--library-print-debug",
+        dest="LibraryPrintDebug",
+        type=bool,
+        default=True,
+        action=DeprecatedOption,
+        help="(Deprecated)",
+    )
```
Comment (ellosel) [inline Tensile/TensileCreateLib/ParseArguments.py:145]: We can remove this altogether.

File: Tensile/TensileCreateLib/__init__.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+################################################################################
+#
+# Copyright (C) 2016-2023 Advanced Micro Devices, Inc. All rights reserved.
```
Comment (ellosel) [inline Tensile/TensileCreateLib/__init__.py:3]: 2024 on files that are new?

File: Tensile/TensileCreateLibrary.py [hunk 5]
```diff
@@ -1440,234 +1439,34 @@ def writeMasterFile(
...
-
-    arguments["CpuThreads"] = args.CpuThreads
-    arguments["PrintLevel"] = args.PrintLevel
-    arguments["IgnoreAsmCapCache"] = args.IgnoreAsmCapCache
-    arguments["WriteMasterSolutionIndex"] = args.WriteMasterSolutionIndex
-
-    for key, value in args.global_parameters:
-        arguments[key] = value
-
-    globalParameters["PrintLevel"] = arguments["PrintLevel"]
```
Comment (ellosel) [inline Tensile/TensileCreateLibrary.py:1657]: Sadly we cannot remove this.

File: Tensile/TensileCreateLib/ParseArguments.py [hunk 6]
```diff
@@ -0,0 +1,312 @@
+################################################################################
+#
+# Copyright (C) 2016-2023 Advanced Micro Devices, Inc. All rights reserved.
```
Comment (ellosel) [inline Tensile/TensileCreateLib/ParseArguments.py:3]: 2024?

**Comment by ellosel** [comment]
Can we add the new files as formatted files during the format check?

File: Tensile/TensileCreateLib/ParseArguments.py (hunk 3 above)
Comment (bstefanuk) [reply Tensile/TensileCreateLib/ParseArguments.py:145]: Also removing `--no-library-print-debug`

File: Tensile/TensileCreateLibrary.py (hunk 5 above)
Comment (bstefanuk) [reply Tensile/TensileCreateLibrary.py:1657]: Moved this call directly into `assignGlobalParameters`

Comment (ellosel) [reply Tensile/TensileCreateLibrary.py:1657]: I don't know if this will preserve behavior. What happens to the header prints?

**Review by ellosel — APPROVED** [review]
Looks good!

**Pull request merged**